middleware.go

```diff
@@ -307,6 +307,13 @@ func addCredentials(c *Client, r *Request) error {
 	return nil
 }
 
+func createCurlCmd(c *Client, r *Request) (err error) {
```

Comment:
@ahuigo Please remove this middleware.

Reply:
**Deleting this middleware will cause `GenerateCurlCommand` to be unable to get the body** (because resty will delete the body after executing the request)

```
	// 3. Generate curl body
	if req.Body != nil {
		buf, _ := io.ReadAll(req.Body)
		req.Body = io.NopCloser(bytes.NewBuffer(buf)) // important!!
		curl += `-d ` + shellescape.Quote(string(buf))
	}

```
Considering that I need to record **body**  before the request is executed(only when `EnableTrace()` is called), 
I need to read body in middleware.

```
func createCurlCmd(c *Client, r *Request) (err error) {
	if r.trace {
		if r.resultCurlCmd==nil{
			r.resultCurlCmd = new(string)
		}
		*r.resultCurlCmd = buildCurlRequest(r.RawRequest, c.httpClient.Jar)
	}
	return nil
}
```

Do you have any other better suggestions?

Reply:
I think the best time to record the body is when the middleware request occurs. If it is before the middleware, parameters such as url will be lost. If it is after the middleware, the body parameters will be lost.

Reply:
@ahuigo I understand. For now, keep this conditional execution for `createCurlCmd` method with `r.trace` enabled. I will see what could be improved in v3.